source/sundials/ida_interface.cc

```diff
@@ -0,0 +1,589 @@
+                     N_Vector yp,
+                     N_Vector resp,
+                     N_Vector tmp1,
+                     N_Vector tmp2,
+                     N_Vector tmp3)
+    {
+      (void) tmp1;
+      (void) tmp2;
+      (void) tmp3;
+      (void) resp;
+      IDAInterface<VEC> &solver = *static_cast<IDAInterface<VEC> *>(IDA_mem->ida_user_data);
+
+      std::shared_ptr<VEC> src_yy = solver.create_new_vector();
+      std::shared_ptr<VEC> src_yp = solver.create_new_vector();
+
+      copy(*src_yy, yy);
+      copy(*src_yp, yp);
+
+      int err = solver.setup_jacobian(IDA_mem->ida_tn,
+                                      *src_yy,
+                                      *src_yp,
+                                      IDA_mem->ida_cj);
+      return err;
+    }
+
+
+    template<typename VEC>
+    int t_dae_solve(IDAMem IDA_mem,
+                    N_Vector b,
+                    N_Vector weight,
+                    N_Vector yy,
+                    N_Vector yp,
+                    N_Vector resp)
+    {
+      (void) weight;
+      (void) yy;
+      (void) yp;
+      (void) resp;
+      IDAInterface<VEC> &solver = *static_cast<IDAInterface<VEC> *>(IDA_mem->ida_user_data);
+
+      std::shared_ptr<VEC> dst = solver.create_new_vector();
+      std::shared_ptr<VEC> src = solver.create_new_vector();
+
+      copy(*src, b);
+
+      int err = solver.solve_jacobian_system(*src,*dst);
+      copy(b, *dst);
+      return err;
+    }
+
+  }
+
+  template <typename VEC>
+  IDAInterface<VEC>::IDAInterface(const std::string name,
+                                  const MPI_Comm mpi_comm) :
+    initial_time(0.0),
+    final_time(1.0),
+    initial_step_size(1e-4),
+    min_step_size(1e-6),
+    abs_tol(1e-6),
+    rel_tol(1e-5),
+    max_order(5),
+    output_period(.1),
+    ignore_algebraic_terms_for_errors(true),
+    ic_type("use_y_diff"),
+    reset_type("use_y_dot"),
+    ic_alpha(.33),
+    ic_max_iter(5),
+    max_non_linear_iterations(10),
+    is_initialized(false),
+    verbose(true),
+    use_local_tolerances(false),
+    ida_mem(nullptr),
+    communicator(Utilities::MPI::duplicate_communicator(mpi_comm)),
+    pcout(std::cout, Utilities::MPI::this_mpi_process(mpi_comm)==0)
+  {
+    set_functions_to_trigger_an_assert();
+  }
+
+  template <typename VEC>
+  IDAInterface<VEC>::~IDAInterface()
+  {
+    if (ida_mem)
+      IDAFree(&ida_mem);
+    MPI_Comm_free(&communicator);
+  }
+
+  template <typename VEC>
+  void IDAInterface<VEC>::add_parameters(ParameterHandler &prm)
+  {
+    prm.add_parameter("Initial step size",initial_step_size);
+
+    prm.add_parameter("Min step size", min_step_size);
+
+    prm.add_parameter("Absolute error tolerance", abs_tol);
+
+    prm.add_parameter("Relative error tolerance", rel_tol);
+
+    prm.add_parameter("Initial time", initial_time);
+
+    prm.add_parameter("Final time", final_time);
+
+    prm.add_parameter("Time units between each output", output_period);
```

Comment:
not being a native speaker, i perceive `units` as units of time (second, hours, etc), thus the description is confusing  IMO. Maybe change to `Time interval between each output`?